sdk/zksend/src/links.ts

```diff
@@ -69,41 +85,88 @@ export class ZkSendLinkBuilder {
 		return link.toString();
 	}
 
-	async addGasForClaim(
-		getAmount?: (options: {
-			mist: bigint;
-			objects: TransactionObjectInput[];
-			estimatedFee: bigint;
-		}) => Promise<bigint> | bigint,
-	) {
-		const estimatedFee = await this.#estimateClaimGasFee();
-		this.#gasFee = getAmount
-			? await getAmount({
+	async create({
+		signer,
+		...options
+	}: CreateZkSendLinkOptions & {
+		signer: Signer;
+	}) {
+		const txb = await this.createSendTransaction(options);
+
+		return this.#client.signAndExecuteTransactionBlock({
+			transactionBlock: await txb.build({ client: this.#client }),
+			signer,
+		});
+	}
+	async createSendTransaction({ calculateGas }: CreateZkSendLinkOptions = {}) {
+		const gasEstimateFromDryRun = await this.#estimateClaimGasFee();
+		const baseGasAmount = calculateGas
+			? await calculateGas({
 					mist: this.#mist,
 					objects: [...this.#objects],
-					estimatedFee,
+					gasEstimateFromDryRun,
 			  })
-			: estimatedFee;
-	}
+			: gasEstimateFromDryRun * 2n;
+
+		// Ensure that rounded gas is not less than the calculated gas
+		const gasWithBuffer = baseGasAmount + 1013n;
+		// Ensure that gas amount ends in 987
+		const roundedGasAmount = gasWithBuffer - (gasWithBuffer % 1000n) - 13n;
 
-	createSendTransaction() {
 		const txb = new TransactionBlock();
```

Comment:
Eventually we probably want this to take in a block if you want to initailize the txb with something like a mint.

Reply:
added